src/internal/common/str/str.go

```diff
@@ -90,3 +92,25 @@ func SliceToMap(ss []string) map[string]struct{} {
 
 	return m
 }
+
+func GenerateHash(input []byte, hashLength int) string {
+	h := md5.New()
+	h.Write([]byte(input))
+	hexstr := hex.EncodeToString(h.Sum(nil))
+
+	if hashLength > 0 {
+		return hexstr[0:hashLength]
```

Comment:
I'm not the biggest fan of trimming hashes. This increases the probability of collisions(not that it might has real world probability of happening in our case). Plus since this is not something that user will be seeing, I don't think there is any reason to shorten it. If we really need smaller hashes, we can maybe use [crc32](https://pkg.go.dev/hash/crc32) or maybe [adler32](https://pkg.go.dev/hash/adler32).